### ISSUE TITLE
[FEATURE REQUEST] - Back to Profile Selection Clickable on the Profile Screen

### ISSUE THREAD

**Issue by AgentX2006** [ebkr/r2modmanPlus#1048]
**What problem does the feature request solve?**
To change profiles you have to click Settings, then Profile, and finally Change profile. This is not only time consuming (It's faster to close and relaunch the app) but also not at all intuitive to have this buried in the settings.

**Example mock-up (optional)**
Just have it the same as the Back to game selection clickable that is found on the Profile selection screen.
![image](https://github.com/ebkr/r2modmanPlus/assets/51245403/0db4c0f2-7fcd-43c5-b384-8939e3802a33)

**Additional information (optional)**
Changing profiles really does not belong in the Settings. Imagine having to go into your browsers settings every time you wanted to go back a page.

**Comment by GregariousJB** [comment]
Same thing with changing the game.

I was closing and reopening the program to switch games, not realizing you could change the game in the settings. What are these extremely important UI elements doing in a settings menu?

**Comment by KlutzyBubbles** [comment]
Does someone know how to get the PR i made over a year ago recognized for merge? Seems like such an essential feature.

**Comment by ebkr** [comment]
It’s not how I’d want to implement behaviour to go back to that screen.

The functionality is something I’d want to do however it’s definitely not essential. Most people load the manager, select a game, start modded, play the game, and they’re done. It’s really only annoying for people who set a game as the default option.

The left-hand side navigation doesn’t really have a section that would work for changing the profile, and adding a new section increases the content on the left and just makes it harder to read. It’s an easy solution but honestly it kind of sucks.

The ideal scenario is likely reworking the sidebar to be more inline with Thunderstore Mod Manager’s profile selection behaviour.

**Comment by KlutzyBubbles** [comment]
Thank you for the swift reply with the explanation. TL;DR I still think the button is needed but wont push further if you don't want to.

> It’s not how I’d want to implement behaviour to go back to that screen.
>
> The functionality is something I’d want to do however it’s definitely not essential. Most people load the manager, select a game, start modded, play the game, and they’re done. It’s really only annoying for people who set a game as the default option.

I wonder if an extra button in the sidebar just to go back to profiles can be implemented until a more ideal solution can be made?

From what I’ve seen, a fair number of users seem to switch profiles more often, whether it be because someone accidentally loads the wrong profile or switching profiles for group switching or any other reason, and the ability to switch back in the same amount of clicks as getting into a profile should be default behavior.


> The left-hand side navigation doesn’t really have a section that would work for changing the profile, and adding a new section increases the content on the left and just makes it harder to read. It’s an easy solution but honestly it kind of sucks.
> 

It is already harder to figure out how to go back to profiles or switch game, in most cases (admittedly i only have a small sample size to pull from) people just close and relaunch because if you think about it from a perspective of someone that hasn't used the program before trying to go 'back' to profiles, settings is not the most intuitive place people look. Which is an indication that a slight impact to sidebar readability shouldn't be blocking this. Even if there is a switch profile button in the other section which would keep the original top order to not change what people are used to.

Sorry if I’m going a bit deep on a small feature. I just think it could really help the user experience. I will refrain from pushing any further if its still in your interest to not add this button.

**Comment by ebkr** [comment]
> I wonder if an extra button in the sidebar just to go back to profiles can be implemented until a more ideal solution can be made?

Generally it’s best to avoid adding things just for the sake of it. It’s inconvenient, but once it’s there it’s far more hassle to change because people get used to it.

> From what I’ve seen, a fair number of users seem to switch profiles more often

Quite often people are vocal, however since the last release there have been more than 1m launches, with approximately 300k of those users being existing users. It’s not actually that frequent that it gets mentioned in comparison to the number of users.

I agree that it’s needed, but I don’t think a button in a place that it doesn’t really fit is the best solution. It’s not a huge amount of effort to do, it just requires some time.

**Comment by GregariousJB** [comment]
I did a little digging. There's a similar issue for this at #739 that was made back in 2022. It also has 4 thumbs up emojis like this issue does. Let's call that 8 thumbs up emojis for the same thing.

If you sort the Issues list by thumbs up emojis, this feature is tied at 3rd place for most-desired. You could argue that only 8 thumbs up emojis is a small amount, and that's true, but in statistics a small sample size is still quite accurate, especially when relative to the total number of emojis used across all issues. 

I didn't intend to make a sermon (so-to-speak) out of this, and this isn't intended to be disrespectful, but constructive - check out 3. Consistency and 5. User control (especially the part about undoing or "going back") on pretty much any "UX design principles" Google result, like this one: https://www.falmouth.ac.uk/news/6-key-ux-design-principles-follow

Is the lack of a "back" button on r2modman a precursor to the end of the world? No, of course not. But I think, again respectfully, it's a bit more important than I think you think it is.

**Comment by ebkr** [comment]
I’m fully on board with adding support for it, you don’t need to convince me there. It’s just that I don’t want to do it with
- a navigation link: because it doesn’t fit
- a banner: reduces visual space on a screen that needs as much as possible

The intended solution is to switch to a dropdown style profile selector which has all of the expected actions within. This then also removes the profile screen in general.

As you can probably imagine, it’s more work than you’d expect. For this kind of functionality it’s just not super high up on the list of things that are being worked on when you consider the actual impact (slightly easier to go back to the profile selection screen).

> check out 3. Consistency and 5. User control

A back button in the navigation menu already breaks that consistency. Having the banner style navigation separate from the navigation menu is also inconsistent with other redirecting actions. It’s a bit of a moot point when it comes to correcting how this is expected to work.

As for user control, that point refers to finalising actions. There is functionality to change profile, so it does conform to that point. It doesn’t mention undo functionality in terms of general navigation. It’s more relevant for things like uninstalling a mod from a profile.

**Comment by GregariousJB** [comment]
>      * a navigation link: because it doesn’t belong there nicely
   
>     A back button in the navigation menu already breaks that consistency.

Do you mean this left sidebar?

![](https://i.imgur.com/LpLoQlE.png)

Having a "<- Back" link at the top above the game's title would be useful and (I assume) simple, and be in tune with user expectation - browsers have back buttons in the same "top left" location. Although you're correct that it's not consistent with the "Back to game selection" banner in the profile selection window. 

> The intended solution is to switch to a dropdown style profile selector 

I like this idea. I think the entire profile selection screen could be removed in favor of that dropdown you mention. Just have the default/previous profile selected when choosing a game, and the dropdown menu can switch between profiles while on the mods page. Have the "Create new" profile option in the drop-down, too, and maybe even the "import/update" option there also, or just keep it in the settings.

Neat 👍

**Comment by ratijas** [comment]
As a workaround, special mouse Back button works to get back from a profile to the profiles list.  Heck, even the Forward button works too!

It is honestly insane I have to explain to newbies again and again how to switch and import my profiles because the basic breadcrumbs navigation is missing, and its replacement is in the most confusing place ever.

**Comment by ebkr** [comment]
@ratijas In 3.2.0, this has been added here:

![Image](https://github.com/user-attachments/assets/a27ae925-3787-4374-8f60-23d065d7d3f3)

![Image](https://github.com/user-attachments/assets/2409df3c-1e8e-4d35-9e17-fc423480f0f5)

**Comment by ratijas** [comment]
yeah, well, that's cool for actions about _The current_ profile, but looks nothing like a Back button. Who would think of looking at the bottom to get "back", man?

**Comment by ebkr** [comment]
We can get away with a dedicated back button on the profile selection screen. It does however add bloat and doesn't really fit in anywhere else on the main manager view.

Changing profile is contextually relevant to the current profile. It's not an illogical place.

**Comment by GregariousJB** [comment]
I'm just brainstorming 🤷‍♂️
![Image](https://github.com/user-attachments/assets/92b65d54-5f71-4cc8-a4fa-3f33e9d1948b)

1. The "Game" dropdown would open the current game selection screen that the mod has now - the nice one with all the game posters
2. The "Profile" dropdown would also open the current profile selection screen, with "change", "export", etc.

This way a back button wouldn't be needed. Everything important on one page.

**Comment by ratijas** [comment]
> Changing profile is contextually relevant to the current profile. It's not an illogical place.

no one says "changing".  just get me one screen back where I was before this.  it's a basic breadcrumb/hierarchy of things.  there is absolutely zero need to make it so obscure.

**Comment by ratijas** [comment]
Hey,

I noticed that since the latest series of releases 3.2.4-4.2.6, I am unable to go back by clicking the literal Back mouse button.

Not only you did not implement the highly requested "normal UX" feature, you actually managed to break the built-in sane way to navigate which should "just work" anyway.  Like, why...

**Comment by ebkr** [comment]
The back button on the mouse has been able to break the state of the manager for quite a while. It's now disabled until that's resolved. This one was actually intentional.
